File: Java/MergeSort.java

public class MergeSort {
	
	static void mergesort(int l,int h,int[] arr) {
		int m=(l)+(h-l)/2;
 		mergesort(l,m,arr);
		mergesort(m+1,h,arr);
		merge(l,m,h,arr);
	}
	static void merge(int l,int m,int h,int[] arr) {
		int k=l,i=l,j=m+1;
		int[] b=new int[arr.length];
		while((k<=m) && (j<=h)) {
			if(arr[k]<=arr[j]) {
				b[i]=arr[k];
				k=k+1;
			}else {
				b[i]=arr[j];
				j=j+1;
			}
			i=i+1;
		}
		i=l;
		if(k>m) {
			for(int z=j;z<h;z++) {
				b[i]=arr[z];
				i=i+1;
			}
		}else {
			for(int x=l;x<=m;x++) {
				b[i]=arr[x];i=i+1;
			}
		}
		for(int s=l;s<h;s++) {
			arr[s]=b[s];
		}
	}
public static void main(String[] args) {
	try {
	int arr[]= {2,3,42,1,3,4,2};
	mergesort(0, arr.length-1, arr);

	for(int i:arr) {
		System.out.print(i+" ");
	}
	}catch(Exception e) {
		e.printStackTrace();
	
	}
	
}
}

File: Python/merge_sort.py
# Github username: RanaiKuruma 
# Aim: To implement merge sort using recursion 
# Date: 4th October 2022

def __merge(a1,a2,a):
    i = 0 
    j = 0 
    k = 0 
    #Transferring the elements of the small array's into the larger array 
    while(i < len(a1) and j < len(a2)):
        if(a1[i] < a2[j]):
            a[k] = a1[i] 
            k += 1 
            i += 1 
        else:
            a[k] = a2[j]            
            k += 1 
            j += 1 

    #Copying the elements of the first small array 
    while(i < len(a1)):
        a[k] = a1[i]            
        k += 1 
        i += 1 

    #Copying the elements of the second small array 
    while(j < len(a2)):
        a[k] = a2[j]        
        k += 1 
        j += 1 

def mergeSort(a):
    n = len(a)
    if(n == 0 or n == 1):
        return -1

    mid = (n) // 2 
    a1 = a[0:mid]        
    a2 = a[mid:]

    mergeSort(a1)
    mergeSort(a2)
    __merge(a1,a2,a)

arr = [int(ele) for ele in input().split()]    
mergeSort(arr)
print(arr)

